### PR TITLE
[Commands] Add additional #peqzone functionality.

### PR DESCRIPTION
**Pull request by Kinglykrab**
- Add #peqzone flagging capabilities so operators don't have to blanket allow #peqzone access to zones.
- Allows you to set a zone's `peqzone` column to `2` and disallow use of `#peqzone` until they have been given the appropriate flag.
- Add #peqzone_flags command to list your #peqzone flags similar to #flags command.
- Add `character_peqzone_flags` table to database and database_schema.h.
- Required SQL update to add the new table.
- Add $client->ClearPEQZoneFlag(zone_id) to Perl.
- Add $client->HasPEQZoneFlag(zone_id) to Perl.
- Add $client->LoadPEQZoneFlags() to Perl.
- Add $client->SendPEQZoneFlagInfo(client) to Perl.
- Add $client->SetPEQZoneFlag(zone_id) to Perl.
- Add client:ClearPEQZoneFlag(zone_id) to Lua.
- Add client:HasPEQZoneFlag(zone_id) to Lua.
- Add client:LoadPEQZoneFlags() to Lua.
- Add client:LoadZoneFlags() to Lua.
- Add client:SendPEQZoneFlagInfo(client) to Lua.
- Add client:SetPEQZoneFlag(zone_id) to Lua.

### CODE REVIEW

File: utils/sql/db_update_manifest.txt [hunk 1]
```diff
@@ -432,6 +432,7 @@
 9176|2022_01_10_checksum_verification.sql|SHOW COLUMNS FROM `account` LIKE 'crc_eqgame'|empty|
 9177|2022_03_06_table_structure_changes.sql|SHOW COLUMNS FROM `pets` LIKE 'id'|empty|
 9178|2022_03_07_saylink_collation.sql|SELECT * FROM db_version WHERE version >= 9178|empty|
+9179|2022_04_30_character_peqzone_flags.sql|SHOW TABLES LIKE 'character_peqzone_flags'|empty|
```
Comment (Akkadius) [inline utils/sql/db_update_manifest.txt:435]: Will need to rebase up with latest

File: utils/sql/git/required/2022_04_30_character_peqzone_flags.sql [hunk 2]
```diff
@@ -0,0 +1,14 @@
+SET NAMES utf8mb4;
+SET FOREIGN_KEY_CHECKS = 0;
+
+-- ----------------------------
+-- Table structure for character_peqzone_flags
+-- ----------------------------
+DROP TABLE IF EXISTS `character_peqzone_flags`;
+CREATE TABLE `character_peqzone_flags`  (
+  `character_id` int NOT NULL DEFAULT 0,
```
Comment (Akkadius) [inline utils/sql/git/required/2022_04_30_character_peqzone_flags.sql:9]: Would prefer if we used `id` here. I know our other tables are all over the place

File: utils/sql/git/required/2022_04_30_character_peqzone_flags.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  `zone_id` int NOT NULL DEFAULT 0,
+  PRIMARY KEY (`character_id`, `zone_id`) USING BTREE
```
Comment (Akkadius) [inline utils/sql/git/required/2022_04_30_character_peqzone_flags.sql:11]: `id` in the key pair as well

File: common/database_schema.h [hunk 4]
```diff
@@ -62,6 +62,7 @@ namespace DatabaseSchema {
 			{"character_pet_buffs",            "char_id"},
 			{"character_pet_info",             "char_id"},
 			{"character_pet_inventory",        "char_id"},
+			{"character_peqzone_flags",        "character_id"},
```
Comment (Akkadius) [inline common/database_schema.h:65]: `id` given previous feedback

File: zone/zoning.cpp [hunk 5]
```diff
@@ -1050,51 +1047,193 @@ void Client::SendZoneFlagInfo(Client *to) const {
...
 
+void Client::SetPEQZoneFlag(uint32 zone_id) {
+	if (HasPEQZoneFlag(zone_id)) {
+		return;
+	}
+
+	peqzone_flags.insert(zone_id);
+
+	auto query = fmt::format(
+		"INSERT INTO character_peqzone_flags (character_id, zone_id) VALUES ({}, {})",
```
Comment (Akkadius) [inline zone/zoning.cpp:1226]: id

**Comment by Akkadius** [comment]
Minor comments, looks dope

File: utils/sql/git/required/2022_04_30_hp_regen_per_second.sql [hunk 6]
```diff
@@ -1 +1 @@
-ALTER TABLE npc_types ADD COLUMN hp_regen_per_second bigint(11) DEFAULT 0 AFTER hp_regen_rate;
+ALTER TABLE npc_types ADD COLUMN hp_regen_per_second bigint(11) DEFAULT 0 AFTER hp_regen_rate;
```
Comment (Akkadius) [inline utils/sql/git/required/2022_04_30_hp_regen_per_second.sql:1]: To keep from getting other merged PR's showing up in your PR you can do a rebase

```
git pull --rebase origin master
```

Merge conflicts if any eg

```
CONFLICT (content): Merge conflict in zone/zonedb.cpp
error: could not apply f22c1321f... Refactor / simplify NPC loading
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
```

Then when fully rebased, push

```
git push --force origin commands/peqzone_flags
```

Comment (Kinglykrab) [reply utils/sql/git/required/2022_04_30_hp_regen_per_second.sql:1]: Gotcha.

**Review by Natedog2012 — APPROVED** [review]

**Pull request merged**